server/http/helloapp/bindings.go

```diff
@@ -12,51 +12,76 @@ import (
+	sendSurvey := h.makeCall(PathSendSurvey)
 
-	httputils.WriteJSON(w,
-		[]*api.Binding{
-			{
-				LocationID: api.LocationCommand,
-				Label:      "hello",
-				Bindings: []*api.Binding{
-					{
-						Label:       "message",
-						Hint:        "[--user] message",
-						Description: "send a message to a user",
-						Call:        call,
-					}, {
-						Label:       "manage",
-						Hint:        "subscribe | unsubscribe ",
-						Description: "manage channel subscriptions to greet new users",
-						Bindings: []*api.Binding{
-							{
-								Label:       "subscribe",
-								Hint:        "[--channel]",
-								Description: "subscribes a channel to greet new users",
-								Call:        h.makeCall(PathMessage, "mode", "on"),
-							}, {
-								Label:       "unsubscribe",
-								Hint:        "[--channel]",
-								Description: "unsubscribes a channel from greeting new users",
-								Call:        h.makeCall(PathMessage, "mode", "off"),
-							},
-						},
-					},
+	c := *sendSurvey
+	sendSurveyModal := &c
+	sendSurveyModal.Type = api.CallTypeForm
+
+	out := []*api.Binding{
+		{
+			// TODO make this a subscribe button, with a state (current subscription status)
+			LocationID: api.LocationChannelHeader,
+			Bindings: []*api.Binding{
+				{
+					LocationID:  "send",
+					Label:       "Survey a user",
+					Icon:        "https://raw.githubusercontent.com/mattermost/mattermost-plugin-jira/master/assets/icon.svg",
+					Hint:        "Send survey to a user",
+					Description: "Send a customized emotional response survey to a user",
+					Call:        sendSurvey, // should be Modal eventually
+				},
+			},
+		}, {
+			LocationID: api.LocationPostMenu,
+			Bindings: []*api.Binding{
+				{
+					LocationID:  "send-me",
+					Label:       "Survey myself",
+					Hint:        "Send survey to myself",
+					Description: "Send a customized emotional response survey to myself",
+					Call:        sendSurvey, // will use ActingUserID by default
 				},
-			}, {
-				LocationID: api.LocationPostMenu,
-				Bindings: []*api.Binding{
-					{
-						LocationID:  "message",
-						Label:       "message",
-						Description: "message a user",
-						Call:        &modal,
+				{
+					LocationID:  "send",
+					Label:       "Survey a user",
+					Hint:        "Send survey to a user",
+					Description: "Send a customized emotional response survey to a user",
+					Call:        sendSurveyModal,
+				},
+			},
+		},
+		// TODO /Command binding is a placeholder, may not be final, test!
+		{
+			LocationID: api.LocationCommand,
+			Bindings: []*api.Binding{
+				{
+					Label:       "message",
+					Hint:        "[--user] message",
+					Description: "send a message to a user",
+					Call:        sendSurvey,
+				}, {
+					LocationID:  "manage",
+					Hint:        "subscribe | unsubscribe ",
+					Description: "manage channel subscriptions to greet new users",
+					Bindings: []*api.Binding{
+						{
+							Label:       "subscribe",
```

Comment:
Here we are using `Label` instead of `LocationID`. Is this intentional?

Reply:
0/5 for subcommands LocationID and Label are always the same, so it'd be nice to avoid duplication, and only require 1 field. My original thinking was to make the Label optional and default it to LocationID. Then we discussed it offline, and flipped it so the LocationID defaults to label. We should probably discuss the semantics of the binding props offline with @larkox.

Reply:
Why don't we stick to them being separate things, Location is required, and _maybe_ default Label to Location if it's empty.